app/models/concerns/translatable.rb

```diff
@@ -47,23 +47,69 @@ def attr_translatable(*attributes)
     end
   end
 
-  def get_translation(attribute)
-    translation = translations.find_by(translatable_attribute: attribute, locale: I18n.locale)
-    translation = get_translations(attribute).first  unless translation
-    return associated_translation(translation) if translation
-    nil
+  def get_translation(attribute, locale: I18n.locale, exact_match: false)
+    # important to filter against the association held in memory instead of performing a db lookup
+    # in order to redisplay potentially transient values
+    result = translations.find do |t|
+      t.translatable_attribute.to_sym == attribute.to_sym && t.locale.to_sym == locale.to_sym
+    end
+    unless result || exact_match
+      # fall back to first translations of any locale if desired locale not present
+      # (except when fetching in order to perform update)
```

Comment:
I don't really get this one either. I don't know what is meant by 'fetching in order to perform update'. How is that captured in the boolean expression on line 56?